plugins/network-elements/bigswitch/src/com/cloud/network/bigswitch/BigSwitchBcfApi.java

```diff
@@ -296,7 +296,7 @@ private void setHttpHeader(final HttpMethodBase m) {
         m.setRequestHeader(CONTENT_TYPE, CONTENT_JSON);
         m.setRequestHeader(ACCEPT, CONTENT_JSON);
         m.setRequestHeader(HTTP_HEADER_INSTANCE_ID, CLOUDSTACK_INSTANCE_ID + "-" + zoneId);
-        if (hash != "" ) {
+        if (! "".equals(hash)) {
```

Comment:
"".equals(null) == false isn't it? I do like your readability argument but had a look at the implementation of isNotEmpty and don't like it. Also this should not keep us tight in discussion, so I will change it, unless you stop me.